File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/imports/JavaImportsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.imports;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaImportsSettings.class)
public interface JavaImportsSettings {

  Optional<Integer> classCountToUseImportWithWildcard();

  Optional<Integer> namesCountToUseStaticImportWithWildcard();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/imports/JavaImportsSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.imports

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaImportsSettings(

    @JsonProperty("Class count to use import with '*'")
    val classCountToUseImportWithWildcard: Int? = null,

    @JsonProperty("Names count to use static import with '*'")
    val namesCountToUseStaticImportWithWildcard: Int? = null
)
